fili-core/src/main/java/com/yahoo/bard/webservice/data/filterbuilders/DruidInFilterBuilder.java

```diff
@@ -0,0 +1,80 @@
+package com.yahoo.bard.webservice.data.filterbuilders;
+
+import com.yahoo.bard.webservice.data.dimension.Dimension;
+import com.yahoo.bard.webservice.data.dimension.DimensionRowNotFoundException;
+import com.yahoo.bard.webservice.druid.model.filter.AndFilter;
+import com.yahoo.bard.webservice.druid.model.filter.Filter;
+import com.yahoo.bard.webservice.druid.model.filter.InFilter;
+import com.yahoo.bard.webservice.druid.model.filter.NotFilter;
+import com.yahoo.bard.webservice.druid.model.filter.OrFilter;
+import com.yahoo.bard.webservice.druid.model.filter.SelectorFilter;
+import com.yahoo.bard.webservice.web.ApiFilter;
+import com.yahoo.bard.webservice.web.FilterOperation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * A DruidInFilterBuilder builds a conjunction of in-filters for each Dimension, where each in-filter corresponds to a
+ * filter term. So, the filter terms on dimension {@code category}:
+ * <p>
+ * {@code category|id-in[finance,sports],category|desc-contains[baseball]}
+ * <p>
+ * are translated into:
+ * <pre>
+ *     {@code
+ *     {
+ *         "type": "in",
+ *         "dimension": "category",
+ *         "values": ["finance", "sports"]
+ *     }
+ *     }
+ * </pre>
+ * Each filter term is resolved independently of the other filter terms.
+ */
+public class DruidInFilterBuilder extends ConjunctionDruidFilterBuilder {
+    private static final Logger LOG = LoggerFactory.getLogger(DruidInFilterBuilder.class);
+
+    @Override
+    protected Filter buildDimensionFilter(Dimension dimension, Set<ApiFilter> filters)
+            throws DimensionRowNotFoundException {
+        LOG.trace("Building dimension filter using dimension: {} \n\n and set of filter: {}", dimension, filters);
+
+        List<Filter> inFilters = new ArrayList<>();
+        for (ApiFilter filter : filters) {
+            ApiFilter normalizedFilter = filter;
+            if (normalizedFilter.getOperation().equals(FilterOperation.notin)) {
+                normalizedFilter = filter.withOperation(FilterOperation.in);
+            }
+            Filter disjunction = new OrFilter(buildSelectorFilters(
+                    dimension,
+                    getFilteredDimensionRows(dimension, Collections.singleton(normalizedFilter))
+            ));
+
+            // transform disjunction(OR) filter to in-filter
+            Filter inFilter = new InFilter(
+                    dimension,
+                    ((OrFilter) disjunction).getFields().stream()
+                            .map(subFilter -> (SelectorFilter) subFilter)
+                            .map(SelectorFilter::getValue)
+                            .distinct()
+                            .collect(Collectors.toList())
+            );
+
+            inFilters.add(normalizedFilter == filter ? inFilter : new NotFilter(inFilter));
```

Comment:
We can do better than this.
This will result in several overlapping in filters for adjacent filters.  Instead we should collect all the positive filters into one collection and all the negative filters into another collection and build up to two Druid In Filters.

This also suggests we could deduplicate productively in the OrFilterBuilder in the same way.